convert2rhel/pkghandler.py

```diff
@@ -507,3 +507,25 @@ def install_additional_rhel_kernel_pkgs(additional_pkgs):
 def is_rhel_kernel_installed():
     installed_rhel_kernels = get_installed_pkgs_by_fingerprint(system_info.fingerprints_rhel, name="kernel")
     return len(installed_rhel_kernels) > 0
+
+
+def clear_yum_versionlock():
+    loggerinst = logging.getLogger(__name__)
+    _VERSIONLOCK_FILE = '/etc/yum/pluginconf.d/versionlock.list'
+    _vers_lock_file = utils.RestorableFile(_VERSIONLOCK_FILE)
+
+    _vers_lock_file.backup()
+
+    if os.path.isfile(_VERSIONLOCK_FILE) and os.path.getsize(_VERSIONLOCK_FILE) > 0:
+        # The file may exist and be empty => no version locks
+        loggerinst.warn("yum versionlock plugin is in use")
+        loggerinst.info("\nIf you choose to continue, all the package version locks will get cleared"
+                        " to not cause the conversion to fail.")
+        utils.ask_to_continue()
+
+        loggerinst.info("Clearing package versions locks...")
+        call_yum_cmd("versionlock clear", print_output=False)
+    else:
+        loggerinst.info("Usage of yum versionlock plugin not detected.")
+
+    _vers_lock_file.restore()
```

Comment:
Backup this file was requested by @bocekm 

Regarding to restore:   agree here is not better place, but, rollback isn't the better place,because it's called once we cancel the process.. and the goal here is:  backup the original lock file, do all process and restore the original file.

Reply:
The goal here is to backup the lock file, and if conversion fails we roll it back. Restoring the file post migration will presumably have invalid entries due to the switch of packages to redhat repo, @bocekm do correct me if I'm wrong about the invalid entries.

@fellipeh, what you're doing here is backing up the file, clearing the file, then restoring it - all in the same function. This means the lock file is not changed and the function has no side effects except for touching the file. `yum versionlock clear` only clears the file, it does not do anything else. Yum itself checks this file whenever it handles packages whether a package it's handling is locked or not.

Reply:
Yeah, the file should not be restored after the conversion. Since every package reinstalled during the conversions, there's a huge chance the restored version lock would not be valid anymore.
So, as Eric wrote, restore it only during the rollback.